crates/settlement-clients/ethereum/src/lib.rs

```diff
@@ -84,6 +99,52 @@ impl SettlementClient for EthereumSettlementClient {
         Ok(format!("0x{:x}", tx_receipt.transaction_hash))
     }
 
+    async fn update_state_blobs_and_blob(
+        &self,
+        program_output: Vec<[u8; 32]>,
+        kzg_proof: [u8; 48],
+        state_diff: Vec<Vec<u8>>,
+    ) -> Result<String> {
+        let trusted_setup = KzgSettings::load_trusted_setup_file(Path::new("./trusted_setup.txt"))
+            .expect("issue while loading the trusted setup");
+        let (sidecar_blobs, sidecar_commitments, sidecar_proofs) = prepare_sidecar(&state_diff, &trusted_setup).await?;
+        let sidecar = BlobTransactionSidecar::new(sidecar_blobs, sidecar_commitments, sidecar_proofs);
+
+        let eip1559_est = self.provider.estimate_eip1559_fees(None).await?;
```

Comment:
but this is a 4844 txn, won't this be incorrect?

Reply:
no, so eip 1559 is gas fee calculating standard of etheruem.